src/headless-cms/components/CmsPageSearch/CmsPageSearch.tsx

```diff
@@ -0,0 +1,179 @@
+import { Card, SearchInput } from 'hds-react';
+import Link from 'next/link';
+// import { useRouter } from 'next/router';
+import React from 'react';
+
+import { ROUTES } from '../../../domain/app/routes/constants';
+import { getEventPlaceholderImage } from '../../../domain/event/utils';
+import {
+  Page,
+  PageIdType,
+  usePagesSearchQuery,
+  useSubPagesSearchQuery,
+} from '../../../generated/graphql-cms';
+import { useCMSClient } from '../../cmsApolloContext';
+import styles from './cmspagesearch.module.scss';
+
+const CmsPageSearch: React.FC<{
+  page?: Page | undefined | null;
+}> = ({ page }): JSX.Element => {
+  if (page) {
+    return <CmsPageSearchFromSubPages page={page} />;
+  }
+  return <CmsPageSearchFromAllPages />;
+};
+
+/**
+ * Search a page from all the pages.
+ */
+const CmsPageSearchFromAllPages: React.FC = (): JSX.Element => {
+  const [searchResult, setSearchResult] = React.useState<Page[]>([]);
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const cmsClient = useCMSClient();
+
+  // Search from all the pages
+  const { data: pageData, loading } = usePagesSearchQuery({
+    client: cmsClient,
+    variables: {
+      search: searchTerm ?? '',
+    },
+  });
+
+  // Update the search results
+  React.useEffect(() => {
+    if (!loading && pageData?.pages) {
+      const pages = pageData?.pages?.edges
+        ?.map((edge) => edge?.node as Page)
+        .filter((page) => !!page);
+      setSearchResult(pages ?? []);
+    }
+  }, [searchTerm, pageData, loading]);
+
+  return (
+    <div>
+      <CmsPageSearchForm setSearchTerm={setSearchTerm} />
+      <CmsPageSearchList pages={searchResult} />
+    </div>
+  );
+};
+
+/**
+ * Search a page from page's sub pages.
+ */
+const CmsPageSearchFromSubPages: React.FC<{
+  page: Page | undefined | null;
+}> = ({ page }): JSX.Element => {
+  const [searchResult, setSearchResult] = React.useState<Page[]>([]);
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const cmsClient = useCMSClient();
+
+  // Search sub pages
+  const { data: pageData, loading } = useSubPagesSearchQuery({
+    client: cmsClient,
+    skip: !page?.uri,
+    variables: {
+      id: page?.uri ?? '',
+      idType: PageIdType.Uri,
+      search: searchTerm ?? '',
+    },
+  });
+
+  // Update the search results
+  React.useEffect(() => {
+    if (!loading && pageData?.page) {
+      const subPages = pageData?.page?.children?.edges
+        ?.map((edge) => edge?.node as Page)
+        .filter((page) => !!page);
+      setSearchResult(subPages ?? []);
+    }
+  }, [searchTerm, pageData, loading]);
```

Comment:
is useEffect and separate state for searchResults actually needed here? COuld this be just:

```
  const subPages = pageData?.page?.children?.edges
        ?.map((edge) => edge?.node as Page)
        .filter((page) => !!page);
```

Reply:
Maybe not. I removed it and it seems to work ok.